R/fromRes.R

```diff
@@ -3,29 +3,44 @@
 
 ### called by getRestuls() and plotICC() ---------------------------------------
 
-eapFromRes <- function ( resultsObj, idVarName = NULL, verbose = TRUE ) {
-  eapRo<- setdiff(intersect( which(resultsObj[,"par"] == "eap"),which(resultsObj[,"indicator.group"] == "persons")), which(resultsObj[,"derived.par"] == "rel"))
-  id   <- unique(resultsObj[intersect(which(resultsObj[,"type"] == "tech"), which(resultsObj[,"par"] == "ID")),"derived.par"])
-  id   <- getIdVarName(id, idVarName, verbose=verbose)
-  if ( length ( eapRo ) == 0 ) {
+eapFromRes <- function (resultsObj, idVarName = NULL, verbose = TRUE){
+  checkmate::assert_data_frame(resultsObj)
```

Comment:
This should not be just any data.frame, but exactly the one generated by `getResults()`. I think it would be a better idea to specify an S3 class for the returned object of `getResults()` and then check whether the argument `resultsObj` is an object of this class. The only problem is: if you implement this, the `eapFromRes()` function would fail if you read in old results objects that were created with older package versions. You would no longer have backwards compatibility. That's why I wouldn't actually do anything here. The assertion could also be left in, but would probably be of little use.